### PR TITLE
Remove shared_ptr from DependencyGraph

### PR DESCRIPTION
**Pull request by havogt**
Co-authored-by: Stagno <giacomo.serafini@mail.polimi.it>

## Technical Description

Remove the shared_ptr from DependencyGraph. Preparation for #797.

### CODE REVIEW

**Comment by havogt** [comment]
launch jenkins

**Comment by havogt** [comment]
launch jenkins

File: dawn/src/dawn/IIR/DoMethod.h [hunk 1]
```diff
@@ -78,14 +72,14 @@ class DoMethod : public IIRNode<Stage, DoMethod, void> {
   Interval& getInterval();
   const Interval& getInterval() const;
   inline unsigned long int getID() const { return id_; }
-  const std::shared_ptr<DependencyGraphAccesses>& getDependencyGraph() const;
+  const std::optional<DependencyGraphAccesses>& getDependencyGraph() const;
```
Comment (jdahm) [inline dawn/src/dawn/IIR/DoMethod.h]: This is much cleaner!

**Comment by Stagno** [comment]
launch jenkins

**Comment by havogt** [comment]
launch jenkins

**Comment by Stagno** [comment]
launch jenkins

File: dawn/src/dawn/Optimizer/PassFieldVersioning.cpp [hunk 2]
```diff
@@ -124,14 +123,14 @@ bool PassFieldVersioning::run(
...
 
           const auto& stmt = doMethod.getAST().getStatements()[stmtIndex];
-          newGraph->insertStatement(stmt);
+          newGraph.insertStatement(stmt);
 
           // Try to resolve race-conditions by using double buffering if necessary
-          auto rc = fixRaceCondition(stencilInstantiation, newGraph.get(), stencil, doMethod,
-                                     loopOrder, stageIdx, stmtIndex);
+          auto rc = fixRaceCondition(stencilInstantiation, newGraph, stencil, doMethod, loopOrder,
+                                     stageIdx, stmtIndex); // TODO fix ptr
```
Comment (mroethlin) [inline dawn/src/dawn/Optimizer/PassFieldVersioning.cpp]: is this still TODO?

Comment (Stagno) [reply dawn/src/dawn/Optimizer/PassFieldVersioning.cpp]: removed TODO

**Pull request merged**